test/e2e/split/main.go

```diff
@@ -0,0 +1,225 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"math"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+type options struct {
+	numChunks  int
+	printChunk int
+	printDebug bool
+	writer     io.Writer
+}
+
+func main() {
+	opts := options{
+		writer: os.Stdout,
+	}
+	flag.IntVar(&opts.numChunks, "chunks", 1, "Number of chunks to create focus regexps for")
+	flag.IntVar(&opts.printChunk, "print-chunk", 0, "Chunk to print a regexp for")
+	flag.BoolVar(&opts.printDebug, "print-debug", false, "Print all spec prefixes in non-regexp format. Use for debugging")
+	flag.Parse()
+
+	dir := flag.Arg(0)
+	if dir == "" {
+		exitIfErr(fmt.Errorf("test directory required as the argument"))
+	}
+
```

Comment:
Do we need validation around the opts.printChunk >= opts.numChunks?

Reply:
Disregard: I see that we have validation further down in the call stack. Any reason we could've bubble that validation call  to the driver function?

Reply:
Moved to main.